src/application/contribution-case/contribution-case.controller.js

```diff
@@ -0,0 +1,104 @@
+import { getTranslationFunctions } from "../../utils/get-translations-locale.js";
+import { handleResponse } from "../../utils/handle-response.js";
+import { logger } from "../../utils/logger.js";
+import { Contribution } from "./contribution-case.model.js";
+import { Attachment } from "../attachment/attachment.model.js";
+import mongoose from "mongoose";
+import { StatusCodes } from "http-status-codes";
+
+export const getContributions = async (req, res) => {
+  const LL = getTranslationFunctions(req.locale);
+  try {
+    logger.info("Getting contributions");
+    const { page = 1, limit = 10 } = req.query;
+
+    const [total, contributions] = await Promise.all([
+      Contribution.countDocuments(),
+      Contribution.find()
+        .populate("user_id case_id attachments")
+        .skip((page - 1) * limit)
+        .limit(limit),
+    ]);
+
+    res.status(StatusCodes.OK).json({
+      data: contributions,
+      message: LL.CONTRIBUTION_CASE.CONTROLLER.GET_CONTRIBUTION_SUCCESS(),
+      total,
+      page,
+      limit,
+    });
+
+    logger.info("Successfully got contributions");
+  } catch (error) {
+    logger.error("Failed to get contributions. Error of type: " + error.name);
+    handleResponse(res, error, LL);
+  }
+};
+
+export const createContribution = async (req, res) => {
+  const LL = getTranslationFunctions(req.locale);
+  const session = await mongoose.startSession();
+  await session.startTransaction();
+  try {
+    logger.info("Creating contribution");
+
+    const { user_id, case_id, content, filepaths } = req.body;
+
+    let dbAttachment;
+
+    if (filepaths) {
+      dbAttachment = new Attachment({
+        filepaths,
+      });
+
+      await dbAttachment.save();
+    }
```

Comment:
Aqui te hace falta eliminar la data de stale content, ve public cases, de igual manera te dejo aqui un ejemplo, pero no se si vaya funcionar, revisalo bien

```suggestion
if (filepaths) {
      const deleted = await StaleContent.deleteMany({
        filepath: { $in: filepaths },
      });

      if (deleted.deletedCount !== filepaths.length) {
        throw new PublicCaseFailedToUploadImagesError(
          LL.PUBLIC_CASE.ERROR.FAILED_UPLOAD_IMAGES(),
        );
      }

      dbAttachment = new Attachment(
        cleanObject({
          filepaths,
        }),
      );

      await dbAttachment.save();
    }
```